Core/Core/Profile/ProfileViewController.swift

```diff
@@ -19,24 +19,29 @@ import UIKit
 private let drawerTransitioningDelegate = DrawerTransitioningDelegate()
 
 public protocol ProfilePresenterProtocol: class {
-    var view: ProfileViewController? { get set }
+    var view: ProfileViewControllerProtocol? { get set }
     var cells: [ProfileViewCell] { get }
     func didTapVersion()
+    func viewIsReady()
 }
 
+public protocol ProfileViewControllerProtocol: class {
+    func reload()
+    func show(_ route: Route, options: Core.Router.RouteOptions?)
+    func show(_ route: String, options: Core.Router.RouteOptions?)
```

Comment:
Hmm, I would have thought routing should be done in the presenter rather than the view controller protocol.

Reply:
The profile view controller has to dismiss itself prior to the presenter being able to route to anything. Because we  don't want a hard dependency on the view controller, the protocol must either define the dismiss method or, as it was done in bridge, the view controller has defined `show` methods that takes care of dismissing itself and routing.

Reply:
ok, I think that makes sense, that the profile is an exception to the usual pattern.

Reply:
Yeah kind of sucks but someway we have to expose some way to the presenter to handle the dismiss.